yoti-sdk-api/src/main/java/com/yoti/api/client/spi/remote/call/ProfileResponse.java

```diff
@@ -33,28 +37,46 @@ public void setReceipt(Receipt receipt) {
         this.receipt = receipt;
     }
 
+    public ErrorDetails getError() {
+        return error;
+    }
+
+    public void setError(ErrorDetails error) {
+        this.error = error;
+    }
+
     @Override
     public boolean equals(Object o) {
-        if (this == o) return true;
-        if (!(o instanceof ProfileResponse)) return false;
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
 
         ProfileResponse that = (ProfileResponse) o;
-
-        if (!sessionData.equals(that.sessionData)) return false;
-        return receipt.equals(that.receipt);
-
+        if (!sessionData.equals((that.sessionData))) {
+            return false;
+        }
+        if (!receipt.equals((that.receipt))) {
+            return false;
+        }
+        return error.equals(that.error);
```

Comment:
Never return a boolean as a result of a if condition!
```java
return sessionData.equals(that.sessionData) 
        && receipt.equals(that.receipt) 
        && error.equals(that.error);
```

Also AFAICT there's no assurance that any of these properties are not `null` so the comparison should be done using `Objects.equals`, e.g. `Objects.equals(sessionData, that.sessionData)`